Data.m

```diff
@@ -0,0 +1,89 @@
+classdef Data
+    %DATA this class loads and holds the IMU data
+    %   This class wraps the loading and handling of the IMU data
+
+    properties
+        IMU_data
+        windowSize
+        windowInterval
+    end
+
+    methods
+        function obj = Data(pattern, windowSize, interval)
+            arguments
+                pattern string = "IMU_Data_for_assignment/**/*.dat";
+                windowSize int16 = 1;
+                interval int16 = 1;
+            end
+            %Data Construct an instance of this class
+            
+            obj.windowInterval = interval;
+            obj.windowSize = windowSize;
+
+            files = dir(pattern);
+            columns = [];
```

Comment:
Variable size changes every loop, pre-allocate and then shrink as needed instead to be more efficient